File: ws/gavilanez/u2/ws24Zoo/WS24Zoo/src/main/java/ec/edu/espe/zoo/model/Eagle.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package ec.edu.espe.zoo.model;

import java.util.Date;

/**
 *
 * @author Kenny Gavilanez, Dev Dynasty, DCCO-ESPE
 */
public class Eagle extends Bird{

    public Eagle(int id, String description, Date dateOnborn, char gender, int numberOfBones, Cage cage) {
        super(id, description, dateOnborn, gender, numberOfBones, cage);
    }

    

    @Override
    public void Feed(int food) {
        System.out.println("feeding an eagle with " + food + "mice");
    }
    
}

File: hw/ccedeno/u2/HW18ZooInPythonJavaScript/ZooInJavaScript/ec.edu.espe.Zoo.model/Eagle.js
const Bird = require('../ec.edu.espe.Zoo.model/Bird');

class Eagle extends Bird {
  constructor(id, description, dateOnborn, gender, numberOfBones, cage) {
    super(id, description, dateOnborn, gender, numberOfBones, cage);
  }

  feed(food) {
    throw new Error('Method "feed" is not supported yet.');
  }
}

module.exports = Eagle;
